### DISCUSSION TITLE
No issues detected in C# analysis

### DISCUSSION THREAD

**Discussion by SR4ven** [jenkinsci/sonar-gerrit-plugin#152] [Q&A] [answered]
### Jenkins and plugins versions report

Jenkinss Version: 2.426.1
Sonar Gerrit Version: 385.v890738eeff15

### What Operating System are you using (both controller, and any agents involved in the problem)?

Windows on controller and agent

### Reproduction steps

1. Analyse a C# project
2. Wait for Sonar Gerrit to post the result

### Expected Results

The issues that are detected during the analysis and that are shown in the SonarQube pull request UI should be reported to Gerrit.

### Actual Results

No issues are detected:
```
SonarQube task '...' completed.
Issues to be commented: 0
Issues to be involved in score calculation: 0
Review has been sent
```

### Anything else?

This is not a permission problem, the user running the analysis has browse permissions on the project and can see the issues. This has been verified in the SonarQube access.log

The reporting is working in other projects with e.g. C++.

My guess is that the problem lies in the file paths that SonarQube reports (`component` in API), which differ from the file path of the git repo.

Calling `api/issues/search?componentKeys=test-project&pullRequest=1234` returns something like this:
```json
"issues": [
    {
        "key": "...",
        "rule": "csharpsquid:S112",
        "severity": "MAJOR",
        "component": "test_project:App/Services/CredentialItemProvider.cs",
        "project": "test-project",
```
However, the full file path from the root of the git repo is `Sources/App/Services/CredentialItemProvider.cs`.
So SonarQubes C# scanner apparently analyses each project separately and changes the base dir.
Setting `sonar.sources` and `sonar.projectBaseDir` to the repo root does not help.

Could this be why the plugin does not detect any issues?
How can this be fixed?

Thanks in advance

**Comment by calisto-le** [comment, marked as answer]
The issue was resolved. There were two misconfigurations:

First was to get the file path of the SonarQube to match the folder structure of the actual git repo. Providing the full qualified path name of the Visual Studio Solution to msbuild resulted in correct SonarQube paths.

Second issue that got the plugin working for the .NET analysis was using a full git clone instead of shallow clone.

The working build script looks like:
`MSBuild.exe %~dp0test_project.sln -t:rebuild`

Also remember to set the MSBuild SonarScanner property for the projectBaseDir to project root. This is the Jenkins Workspace in this case.

`/d:sonar.projectBaseDir=%WORKSPACE%`
